src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -138,8 +139,9 @@ public void SslStream_StreamToStream_Alpn_NonMatchingProtocols_Fail()
                     ServerCertificate = certificate,
                 };
 
-                Assert.ThrowsAsync<AuthenticationException>(() => { return client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None); });
-                Assert.ThrowsAsync<AuthenticationException>(() => { return server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None); });
+                Assert.True(DoHandshakeWithOptions(client, server, clientOptions, serverOptions));
+                Assert.Equal(default, client.NegotiatedApplicationProtocol);
```

Comment:
This behavior is wrong, if the client and server fail to match alpn, it should fail the handshake as per the rfc. Why did you change this behavior?

Reply:
I didn't change the behaviour, see above I don't believe the test was ever working.

Reply:
once I figure out what is going on, I will just disable the tests with an issue.

Reply:
Uhh, that's a bug, it should be async () => await <stuff>

Reply:
I addressed this in the cancellation PR #24849 You can undo these changes here.

Reply:
Are you sure that fixes it, don't you need to await the result of ThrowAsync<T> from memory the code is basically 

```
public async static Task<T> ThrowsAsync<T>(Func<Task> testCode) where T : Exception
{
    try
    {
        await testCode();
        Assert.Throws<T>(() => { }); // Use xUnit's default behavior.
    }
    catch (T exception)
    {
        return exception;
    }
    return null;
}
```

without the 

``` await Assert.ThrowAsync(````
it still won't wait for the assert? Or am I wrong on that?

https://github.com/dotnet/corefx/blob/master/src/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs#L66

Reply:
i've been trying to get the implementation of ThrowsAsync, thanks for getting it. Yeah it looks like the throwsasync needs to be awaited.

Reply:
so when i do, the windows tests hang, and the linux tests fail.I will disable with an issue and check them in.

Reply:
Ok sounds good! thanks!

Reply:
Just letting them run with async to make sure that need to be disabled on everything or just one OS. Then I will tidy up with an [Issue] tag